### ISSUE TITLE
Bitbucket server http support

### ISSUE THREAD

**Issue by RishabhTayal** [danger/danger#784]
# Report
Currently bitbucket server integration doesn't support http hosts. The url has to be https (as seen here [bitbucket_server_api.rb](https://github.com/danger/danger/blob/master/lib/danger/request_sources/bitbucket_server_api.rb#L14)). Would be nice if we pass the `DANGER_BITBUCKETSERVER_HOST` with `http://` or `https://`. 

## What did you do?

Setting up `danger` on Jenkins with BBS.

## What did you expect to happen?

run Danger without error.

## What happened instead?

Got error while running.

```
bundler: failed to load command: danger (bin/danger)
OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv2/v3 read server hello A: unknown protocol
```

## Your Environment

* Which CI are you running on? *Jenkins*
* Are you running the latest version of Danger? *Yes*
* What is your Dangerfile?

  ```ruby
  declared_trivial = bitbucket_server.pr_title.include? "#trivial"

  warn("PR is classed as Work in Progress") if bitbucket_server.pr_title.include? "[WIP]"

  warn("Big PR") if git.lines_of_code > 500

  fail("fdescribe left in tests") if `grep -r fdescribe specs/ `.length > 1
  fail("fit left in tests") if `grep -r fit specs/ `.length > 1
  ```

**Comment by orta** [comment]
Sure, I'd be happy to have this in 👍

**Comment by RishabhTayal** [comment]
Here. Took a shot. https://github.com/danger/danger/pull/785

**Comment by orta** [comment]
This should be fixed by 5.0.2

**Comment by RishabhTayal** [comment]
It works now in 5.0.2. Just tested. 👍
